jenkins/debian/debos/scripts/crush.sh

```diff
@@ -1,156 +1,202 @@
+    # drop gcc-6 python helpers
+    rm -rf usr/share/gcc-6
 
-# No need for fuse
-find usr etc -name '*fuse*' -prune -exec rm -r {} \;
+    # Drop sysvinit leftovers
+    rm -rf etc/init.d
+    rm -rf etc/rc[0-6S].d
 
-# lsb init function leftovers
-rm -rf usr/lib/lsb
+    # Drop upstart helpers
+    rm -rf etc/init
 
-# Utils using ncurses
-rm -f usr/bin/pg
-rm -f usr/bin/watch
-rm -f usr/bin/slabtop
+    # Various xtables helpers
+    rm -rf usr/lib/xtables
 
-# boot analyser
-rm -f usr/bin/systemd-analyze
+    # Drop all locales
+    # TODO: only remaining locale is actually "C". Should we really remove it?
+    rm -rf usr/lib/locale/*
 
-# Only needed when adding libraries
-rm -f usr/sbin/ldconfig*
+    # local compiler
+    rm -f usr/bin/localedef
 
-# Games, unused
-rm -rf usr/games
+    # lsb init function leftovers
+    rm -rf usr/lib/lsb
 
-# Unused systemd generators
-rm -f lib/systemd/system-generators/systemd-cryptsetup-generator
-rm -f lib/systemd/system-generators/systemd-debug-generator
-rm -f lib/systemd/system-generators/systemd-gpt-auto-generator
-rm -f lib/systemd/system-generators/systemd-hibernate-resume-generator
-rm -f lib/systemd/system-generators/systemd-rc-local-generator
-rm -f lib/systemd/system-generators/systemd-system-update-generator
-rm -f lib/systemd/system-generators/systemd-sysv-generator
+    # boot analyser
+    rm -f usr/bin/systemd-analyze
 
-# Efi blobs
-rm -rf usr/lib/systemd/boot
+    # Only needed when adding libraries
+    rm -f usr/sbin/ldconfig*
 
-# Translation catalogs
-rm -rf usr/lib/systemd/catalog
+    # Games, unused
+    rm -rf usr/games
 
-# Misc systemd utils
-rm -f usr/bin/bootctl
-rm -f usr/bin/busctl
-rm -f usr/bin/hostnamectl
-rm -f usr/bin/localectl
-rm -f usr/bin/systemd-cat
-rm -f usr/bin/systemd-cgls
-rm -f usr/bin/systemd-cgtop
-rm -f usr/bin/systemd-delta
-rm -f usr/bin/systemd-detect-virt
-rm -f usr/bin/systemd-mount
-rm -f usr/bin/systemd-path
-rm -f usr/bin/systemd-run
-rm -f usr/bin/systemd-socket-activate
-
-# Remove pam module to authenticate against a DB
-# plus libdb-5.3.so that is only used by this pam module
-rm -f usr/lib/*/security/pam_userdb.so
-rm -f usr/lib/*/libdb-5.3.so
-
-# remove NSS support for nis, nisplus and hesiod
-rm -f usr/lib/*/libnss_hesiod*
-rm -f usr/lib/*/libnss_nis*
+    # Remove pam module to authenticate against a DB
+    # plus libdb-5.3.so that is only used by this pam module
+    rm -f usr/lib/*/security/pam_userdb.so
+    rm -f usr/lib/*/libdb-5.3.so
+
+    # remove NSS support for nis, nisplus and hesiod
+    rm -f usr/lib/*/libnss_hesiod*
+    rm -f usr/lib/*/libnss_nis*
+}
+
+# Removing unneeded packages
+remove_packages()  {
+   UNNEEDED_PACKAGES="$@"
+
+   for PACKAGE in ${UNNEEDED_PACKAGES}
+   do
+	echo "Forcing removal of ${PACKAGE}"
+	if ! dpkg --purge --force-remove-essential --force-depends "${PACKAGE}"
+	then
+		echo "WARNING: ${PACKAGE} isn't installed"
+	fi
+   done
+}
+
+# Either skip specific components or crush all
```

Comment:
How about a simpler approach by providing the list of functions to call on the command line, and call them all if none is provided?

Say, `./crush.sh` would call all the functions to do the same as it current does.  But `./crush.sh systemd misc_directories` would only call the `systemd()` and `misc_directories()` functions.  That way, it would be clearer which things are actually crushed and it would also simplify the implementation by removing the need for `getopt` and bash arrays.

Reply:
okay will make these changes.

Reply:
Added these changes and tested with debos.